tests/countries/test_sao_tome_and_principe.py

```diff
@@ -0,0 +1,163 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.sao_tome_and_principe import SaoTomeAndPrincipe, ST, STP
+from tests.common import CommonCountryTests
+
+
+class TestSaoTomeAndPrincipe(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(2014, 2050)
+        super().setUpClass(SaoTomeAndPrincipe, years=years, years_non_observed=range(2014, 2050))
```

Comment:
Based on:
```python
        # Holidays are observed on the next Monday if on Sunday, or previous Friday if on Saturday.
        # Based on common government practices since at least 2020.
        kwargs.setdefault("observed_since", 2020)
```
```suggestion
        super().setUpClass(SaoTomeAndPrincipe, years=years, years_non_observed=range(2020, 2050))
```